wardrobe/tradable_kitties/src/tests.rs

```diff
@@ -0,0 +1,864 @@
+    let mut output = input.clone();
+    output.price = Some(500);
+
+    let result = TradableKittyConstraintChecker::<0>::UpdateKittyPrice.check(
+        &[Bogus.into()],
+        &[],
+        &[output.into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::BadlyTyped));
+}
+
+#[test]
+fn update_price_bad_output_path_fails() {
+    let input = TradableKittyData::default_tradable_kitty();
+    let mut output = input.clone();
+    output.price = Some(500);
+    let mut input1 = TradableKittyData::default_tradable_kitty();
+    input1.kitty_basic_data.dna = KittyDNA(H256::from_slice(b"superkalifragislisticexpialadoai"));
+
+    let result = TradableKittyConstraintChecker::<0>::UpdateKittyPrice.check(
+        &[input.into(), input1.into()],
+        &[],
+        &[output.into(), Bogus.into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::BadlyTyped));
+}
+
+#[test]
+fn update_price_different_dna_path_fails() {
+    let input = TradableKittyData::default_tradable_kitty();
+    let mut output = input.clone();
+    output.price = Some(500);
+
+    output.kitty_basic_data.dna = KittyDNA(H256::from_slice(b"superkalifragislisticexpialadoai"));
+
+    let result = TradableKittyConstraintChecker::<0>::UpdateKittyPrice.check(
+        &[input.clone().into()],
+        &[],
+        &[output.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::KittyBasicPropertiesAltered)
+    );
+}
+
+#[test]
+fn update_price_duplicate_input_path_fails() {
+    let input = TradableKittyData::default_tradable_kitty();
+    let mut output = input.clone();
+    output.price = Some(500);
+    // check that only 1 instance with duplicate dna is inserted and when 1st is removed , 2nd one fails.
+    let result = TradableKittyConstraintChecker::<0>::UpdateKittyPrice.check(
+        &[input.clone().into(), input.into()],
+        &[],
+        &[output.clone().into(), output.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::DuplicateTradableKittyFound)
+    );
+}
+
+#[test]
+fn update_price_basic_properties_updated_path_fails() {
+    let input = TradableKittyData::default_tradable_kitty();
+    let mut output = input.clone();
+    output.price = Some(500);
+    output.kitty_basic_data.free_breedings += 1;
+
+    let result = TradableKittyConstraintChecker::<0>::UpdateKittyPrice.check(
+        &[input.into()],
+        &[],
+        &[output.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::KittyBasicPropertiesAltered)
+    );
+}
+
+#[test]
+fn update_price_not_updated_path_fails() {
+    let input = TradableKittyData::default_tradable_kitty();
+    let output = input.clone();
+
+    let result = TradableKittyConstraintChecker::<0>::UpdateKittyPrice.check(
+        &[input.into()],
+        &[],
+        &[output.into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::KittyPriceUnaltered));
+}
```

Comment:
It would indeed be weird for a user to do this, but I could see the case for it also being valid. Up to you. I have no preference.

Reply:
I kept it as it is.